src/mpi/coll/transports/gentran/tsp_gentran.c

```diff
@@ -331,6 +364,28 @@ int MPIR_TSP_sched_localcopy(const void *sendbuf, MPI_Aint sendcount, MPI_Dataty
 }
 
 
+/* Transport function that adds a callback type vertex in the graph */
+int MPIR_TSP_sched_cb(MPIR_TSP_cb_t cb_p, void *cb_data, MPIR_TSP_sched_t sched,
+                      int n_in_vtcs, int *in_vtcs)
+{
+    vtx_t *vtxp;
+    int vtx_id;
+
+    /* assign a new vertex */
+    vtx_id = MPII_Genutil_vtx_create(sched, &vtxp);
+
+    vtxp->vtx_kind = MPII_GENUTIL_VTX_KIND__CB;
+    MPII_Genutil_vtx_add_dependencies(sched, vtx_id, n_in_vtcs, in_vtcs);
+
+    /* record the arguments */
+    vtxp->u.cb.cb_p = cb_p;
+    vtxp->u.cb.cb_data = cb_data;
+
+    MPL_DBG_MSG_FMT(MPIR_DBG_COLL, VERBOSE, (MPL_DBG_FDEST, "Gentran: schedule [%d] cb", vtx_id));
+
+    return vtx_id;
+}
+
```

Comment:
this is probably not the time for it, but wouldn't it be possible to implement all vertex callbacks using this function signature? just the stick the arguments in `void *cb_data`.

Reply:
Can you please elaborate on that?
This function only adds a callback vertex into gentran scheduler. It does not actually call the callback. All callback functions are required to have  the same signature (MPIR_TSP_cb_t).

Reply:
What I am trying to say is that `MPIR_TSP_sched_irecv_status` and `MPIR_TSP_sched_cb` are basically the same function. Perhaps all vertexes could be added as generic callbacks. Anyway this is way out of scope for this PR, so lets just drop it.